src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationHistoryService.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.service
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.recommendation.RecommendationHistoryResponse
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.jpa.repository.RecommendationHistoryRepository
+import uk.gov.justice.hmpps.kotlin.sar.HmppsProbationSubjectAccessRequestService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+
+@Transactional
+@Service
+class RecommendationHistoryService(
+  private val recommendationHistoryRepository: RecommendationHistoryRepository,
+) : HmppsProbationSubjectAccessRequestService {
+  override fun getProbationContentFor(
+    crn: String,
+    fromDate: LocalDate?,
+    toDate: LocalDate?,
+  ): HmppsSubjectAccessRequestContent? {
+    val recommendationHistory = if (fromDate != null && toDate != null) recommendationHistoryRepository.findByCrn(crn, fromDate, toDate) else emptyList()
```

Comment:
That looks a bit dangerous.  This may lead a caller to believe that we hold no data when in fact we might.  
The dates should be mandatory as part of the request, then can't be null and this ambiguity goes away.

If you have no control over the controller and/or request object because they are part of the HMPPS Kotlin library, it might be better to throw an exception here if the dates are null.

Reply:
>you have no control over the controller and/or request object because they are part of the HMPPS Kotlin library
 That's exactly it Darren 👍! Anyway, I've gone for the exception approach you suggested.